### PR TITLE
Docs: clarity for migration guide.

### LINKED ISSUE #2112: Write a migration guide for V10
https://alpha.trpc.io/docs/migrate-from-v9-to-v10

Can be split into different parts

The main chunks of this can be found [here](https://github.com/trpc/trpc/blob/next/.tmp/v10-docs.md#interopability-mode-for-old-routers--migration-path): 

95% use-case:

- [x] Transforming old API for backward-compatabiltiy
- [ ] Add your first router in the new way
- [ ] Briefly explain new Router API
- [ ] Briefly ew client API with proxy
- [ ] [...]

The 5%:

- [ ] New Links
- [ ] New subscriptions with breaking changes
- [ ] [...]

### PR DESCRIPTION
**Pull request by anthonyshew**
- Bring clarity to migration page.

Closes #2112

## 🎯 Changes

This PR features major copy edits to the migration page.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Comment by anthonyshew** [comment]
@KATT Let me know if you like this style of writing. You'll find it feels quite familiar if you compare it to Vercel or Next.js docs. 😄 

If this PR sits well with you, I can do a lot more.

**Review by KATT — APPROVED** [review]
This is great.

Whoops, it seem to have been updated a bit -- can you update it?

File: www/docs/migration/migrate-from-v9-to-v10.mdx [hunk 1]
```diff
@@ -1,257 +1,67 @@
...
-    },
-  });
-
-// NEW:
-const appRouter = t.router({
-  greeting: t.procedure
-    .input(z.string())
-    .query(({ input }) => `hello ${input}!`),
-});
-```
```
Comment (KATT) [inline www/docs/migration/migrate-from-v9-to-v10.mdx:97]: I really liked having the old and new side-to-side, how come that was deleted?

Comment (KATT) [reply www/docs/migration/migrate-from-v9-to-v10.mdx:97]: I guess this belongs more in a blog article - was that your thinking? 🤔

Comment (anthonyshew) [reply www/docs/migration/migrate-from-v9-to-v10.mdx:97]: This does indeed belong. I think I cut it as a part of the larger block that was demonstrating too much - but the stanzas you've marked here are certainly worthy.

I'll find a place for these. My first instinct is first section of the page - but will see once I pop it back open on my machine.

File: www/docs/migration/migrate-from-v9-to-v10.mdx [hunk 2]
```diff
@@ -1,257 +1,82 @@
...
-    .router()
-    /* [...] */
-    .interop();
-
-+ export const appRouter = t.mergeRouters(legacyRouter);
-  export type AppRouter = typeof appRouter;
-```
+```ts title="/server.ts"
+import { initTRPC } from '@trpc/server';
+import { legacyRouter } from './legacyRouter.ts';
```
Comment (KATT) [inline www/docs/migration/migrate-from-v9-to-v10.mdx:60]:
```suggestion
import { legacyRouter } from './legacyRouter';
```

File: www/docs/migration/migrate-from-v9-to-v10.mdx [hunk 3]
```diff
@@ -1,257 +1,82 @@
...
+There are [a few features that are not supported by `.interop()`](#limitations-of-interop). We expect nearly all of our users to be able to use `.interop()` to migrate their server side code in only a few minutes. If you are discovering that `.interop()` is not working correctly for you, be sure to [check here](#limitations-of-interop).
+:::caution
 
-## Migration path for the 95%
+## Enabling `interop()` on a v9 router
 
-If you are migrating from V9->V10, the transition is pretty simple for the 95% use-cases.
+Turning your v9 router into a v10 router only takes 10 characters. Add `.interop()` to the end of your v9 router...and you're done with your server code!
 
-**1. Add `.interop()`**
```
Comment (KATT) [inline www/docs/migration/migrate-from-v9-to-v10.mdx:177]: How come you didn't like the 1/2/3/4 guided way of approaching it? In my head, it's easier to follow

**Review by KATT — COMMENTED** [review]
- Love all the text changes
- Missing a bit comparison of side-by-side comparisons
- Unsure why we didn't like the simple 1/2/3 for the guided 95% usecase

**Comment by KATT** [comment]
Incorporated in #2662 but would love more of your typing magic if you are keen on contributing 💞

**Comment by KATT** [comment]
@all-contributors add @anthonyshew for docs

**Comment by anthonyshew** [comment]
Oh, jeez, sorry, I've had a bunch of travel impacting me! I seem to have dropped this. Will pick it back up as soon as I can.

**Comment by KATT** [comment]
> Oh, jeez, sorry, I've had a bunch of travel impacting me! I seem to have dropped this. Will pick it back up as soon as I can.

No stress, I really appreciate your help!

**Pull request closed without merging**